articles/tokens/concepts/refresh-token-rotation.md

```diff
@@ -0,0 +1,74 @@
+
+The following state diagram illustrates how Refresh Token Rotation is used in conjunction with the Authorization Code Flow with PKCE, but the general principle of getting a new refresh token with each exchange applies to all supported flows.
+
+![Refresh Token Rotation State Diagram](/media/articles/tokens/rtr-state-diagram.png)
+
+This means that you can safely use Refresh Tokens to mitigate the adverse effects of browser privacy tools and provide continuous access to end-users without disrupting the user experience.
+
+## Automatic reuse detection
+
+When a client needs a new Access Token, it sends the Refresh Token with the request to Auth0 to get a new token pair. As soon as the new pair is issued by Auth0, the Refresh Token used in the request is invalidated. This helps safeguard your app from replay attacks resulting from compromised tokens.
+
+Without enforcing sender-constraint, it’s impossible for the authorization server to know which actor is legitimate or malicious in the event of a replay attack. Therefore, it’s important that when a previously-used Refresh Token (already invalidated) is sent to the authorization server that the most recently issued Refresh Token is immediately invalidated as well, preventing any Refresh Tokens in the same token family (all Refresh Tokens descending from the original Refresh Token issued for the client) from being used to get new Access Tokens.
+
+For example, consider the following scenario: 
+1. Legitimate Client has **Refresh Token 1**, and it is leaked to or stolen by Malicious Client. 
+2. Legitimate Client uses **Refresh Token 1** to get a new Refresh Token/Access Token pair.
+3. Auth0 returns **Refresh Token 2/Access Token 2**.
+4. Malicious Client then attempts to use **Refresh Token 1** to get an Access Token. Auth0 recognizes that Refresh Token 1 is being reused, and immediately invalidates the Refresh Token family, including **Refresh Token 2**.
+5. Auth0 returns an **Access Denied** response to Malicious Client.
+6. **Access Token 2** expires and Legitimate Client attempts to use **Refresh Token 2** to request a new token pair. Auth0 returns an **Access Denied** response to Legitimate Client.
+7. Re-authentication is required.
+
+![Reuse Detection](/media/articles/tokens/reuse-detection.png)
+
+This protection mechanism works regardless of whether the legitimate client or the malicious client is able to exchange **Refresh Token 1** for a new token pair before the other. As soon as reuse is detected, all subsequent requests will be denied until the user re-authenticates. When reuse is detected, Auth0 captures this event (`ferrt` indicating a failed exchange) in logs to provide visibility for security reviews and audits. This can be especially useful in conjunction with Auth0’s log streaming capabilities.
```

Comment:
When reuse is detected... -> Auth0 captures detected reuse events (`ferrt` indicating a failed exchange) in logs.